### PR TITLE
Detect changeset creation failure when there's no change

### LINKED ISSUE #875: Error message is shown when launching a subsequent call to "docker compose up" even there were no changes
**Description**

Error message is shown when launching a subsequent call to "docker compose up" even there were no changes to docker-compose.yml file and CloudFormation is already COMPLETED.

**Steps to reproduce the issue:**
1. Launch a `docker compose up`
2. Wait for all resources to be provisioned (cli returns and CloudFormation says it was completed)
3. Launch again the same command line with no changes to docker-compose.yml
4. See an error message: 
```
DEBU[0004] Create CloudFormation Changeset
ResourceNotReady: failed waiting for successful resource state
```

**Describe the results you received:**
An error message indicating there was some kind of failure

**Describe the results you expected:**
A message indicating everything is up to date, or indicating nothing has changed since last call.

**Additional information you deem important (e.g. issue happens only occasionally):**
Even after this message, a subsequent call after changing at least something in docker-compose.yml works as expected.

**Output of `docker version`:**

```
(paste your output here)
```

**Output of `docker context show`:**  
You can also run `docker context inspect context-name` to give us more details but don't forget to remove sensitive content.

```
(paste your output here) 
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false
 Plugins:
  app: Docker Application (Docker Inc., v0.8.0)
  buildx: Build with BuildKit (Docker Inc., v0.4.2-tp-docker)
  scan: Docker Scan (Docker Inc., v0.3.4)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 401
 Server Version: 19.03.13
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: active
  NodeID: vfw3bft5qkwxp8f2x8me35t5d
  Is Manager: true
  ClusterID: 5okhxug6ylqnec72ia0fa3ef7
  Managers: 1
  Nodes: 1
  Default Address Pool: 10.0.0.0/8
  SubnetSize: 24
  Data Path Port: 4789
  Orchestration:
   Task History Retention Limit: 5
  Raft:
   Snapshot Interval: 10000
   Number of Old Snapshots to Retain: 0
   Heartbeat Tick: 1
   Election Tick: 10
  Dispatcher:
   Heartbeat Period: 5 seconds
  CA Configuration:
   Expiry Duration: 3 months
   Force Rotate: 0
  Autolock Managers: false
  Root Rotation In Progress: false
  Node Address: 192.168.65.3
  Manager Addresses:
   192.168.65.3:2377
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fba4e9a7d01810a393d5d25a3621dc101981175
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 5
 Total Memory: 7.78GiB
 Name: docker-desktop
 ID: M2XC:7THC:EZRF:QYS5:AXAI:HM3J:QTMA:RF3G:EUNT:OOV7:X2ED:EPYM
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Additional environment details (AWS ECS, Azure ACI, local, etc.):**
ECS

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Add detection of changeset creation failure when compose model has no change vs deployed one

**Related issue**
closes #875

/test-ecs

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

File: ecs/sdk.go [hunk 1]
```diff
@@ -264,9 +264,18 @@ func (s sdk) CreateChangeSet(ctx context.Context, name string, template []byte)
 		return "", err
 	}
 
-	err = s.CF.WaitUntilChangeSetCreateCompleteWithContext(ctx, &cloudformation.DescribeChangeSetInput{
+	s.CF.WaitUntilChangeSetCreateCompleteWithContext(ctx, &cloudformation.DescribeChangeSetInput{ // nolint:errcheck
```
Comment (chris-crone) [inline ecs/sdk.go:267]: Is there no other relevant error that can be returned here?

Comment (ndeloof) [reply ecs/sdk.go:267]: we actually get a false-negative error here. This API returns "ResourceNotReady" error as the changeset failed to  create, but we still need to wait for changeset creation completion before we can inspect it's state

Comment (chris-crone) [reply ecs/sdk.go:267]: Would be good to add a comment about why we're ignoring the error then :)

Comment (ndeloof) [reply ecs/sdk.go:269]: done

**Review by chris-crone — APPROVED** [review]
LGTM

**Pull request merged**